lib/std/Build/Step/WriteFile.zig

```diff
@@ -89,18 +79,36 @@ pub fn create(owner: *std.Build) *WriteFile {
     return write_file;
 }
 
+/// Returns a lazy path representing a file after it has been written.
+/// Asserts that a file named `sub_path` has already been added to this step.
+pub fn getFile(write_file: *WriteFile, sub_path: []const u8) std.Build.LazyPath {
+    for (write_file.files.items) |file| {
```

Comment:
Oops, when I looked at this last night I mistakenly thought the files array was already an ArrayHashMap.

As a general rule of thumb, assertions should not bump up the tier of big O complexity. And as a general rule of thumb, don't make operations O(N) when they could be made effectively O(1) via a hash map data structure.

I believe the correct path forward here is to make the files array into an ArrayHashMapUnmanaged with the sub_path as a key.

`add` and related functionality should gain doc comments that it will assert the same file is not written twice as that is very likely a misconfiguration.

Reply:
Here's two alternative ideas I'd prefer to this latest ArrayHashMap suggestion.

1. (Most preferred) remove `getFile` altogether.  If the caller needs a `LazyPath` for any of the files being written, it uses the LazyPath value that is returned from calling `add`.  Adding this hash map to access the lazy paths after the fact when there's already a way to get them at the time they are created feels a bit too divergent to me.

2. (Second preference) remove `getFile` and go back to the original API (or a slight variation).  The difference here is that instead of each File/Directory having it's own `GeneratedFile`, they could just store a `LazyPath` as a field that is referencing the main generated file.  Then we could have the code using this step just access this new `LazyPath` field directory and/or call `getPath` like they're doing now (`getPath` just returns the new lazy path field).

Reply:
I've pushed a new commit that goes with option 1.